src/sinks/util/udp.rs

```diff
@@ -171,7 +171,17 @@ impl Sink for UdpSink {
                         error!(message = "send failed", %error);
                         Ok(AsyncSink::NotReady(line))
                     }
-                    Ok(_) => Ok(AsyncSink::Ready),
+                    Ok(sent) => {
+                        if sent != line.len() {
+                            warn!(
```

Comment:
@juchiast is this better represented as an event? And should it the log be rate-limited?

Reply:
Good point! We don't have back-off for this event so it is better to rate-limit it. I added an internal event for this. I also changed it to `error!`, not sure why I used `warn!` originally.